app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt

```diff
@@ -245,13 +266,28 @@ class EditBookmarkFragment : Fragment(R.layout.fragment_edit_bookmark) {
                         )
                     )
                 }
+                withContext(Main) {
+                    progress_bar_bookmark.visibility = View.INVISIBLE
+
+                    inputLayoutBookmarkUrl.error = null
+                    inputLayoutBookmarkUrl.errorIconDrawable = null
+
+                    findNavController().popBackStack()
+                }
             } catch (e: UrlParseFailed) {
                 withContext(Main) {
-                    bookmarkUrlEdit.error = getString(R.string.bookmark_invalid_url_error)
+                    progress_bar_bookmark.visibility = View.INVISIBLE
+
+                    inputLayoutBookmarkUrl.error = getString(R.string.bookmark_invalid_url_error)
+                    inputLayoutBookmarkUrl.error
+                    inputLayoutBookmarkUrl.setErrorIconDrawable(R.drawable.mozac_ic_warning_with_bottom_padding)
+                    inputLayoutBookmarkUrl.setErrorIconTintList(
+                        ColorStateList.valueOf(
+                            ContextCompat.getColor(requireContext(), R.color.design_error)
+                        )
+                    )
                 }
             }
         }
-        progress_bar_bookmark.visibility = View.INVISIBLE
```

Comment:
I think we should consider keeping this here since we call it in both 270 and 279 now.